### PR TITLE
feat: relocate hybridse java

### LINKED ISSUE #495: ci: build improve
We use `mkdir; cd && cmake && make &&` in many places. But a list of cmd, e.g. `cmake && make && cd`, if make(the internal cmd of list) got error,  `set -e` can't handle this error.

We should convert to multi-line script, without `&&`, `||`

_Originally posted by @aceforeverd in https://github.com/4paradigm/OpenMLDB/pull/493#discussion_r724812660_

But many places need convert, maybe we could write a good build script.

### LINKED ISSUE #674: feat: move hybridse java module to OpenMLDB
Modules in `java` e.g `opgenmldb-batch` depends on hybridse java module. Since both of the modules are in the same code base, it is necessary to merge hybridse java module into OpenMLDB.

### PR DESCRIPTION
**Pull request by aceforeverd**
- move hybridse java to `java` directory, close #674 
- refactor coverage job, #677 
- minor bug fix for top Makefile

previously, the coverage report only covers code inside `hybridse` directory, specificly, hybridse cpp and hybridse-sdk.
now coverage also include OpenMLDB cpp and java code.

## Limit

+ OpenMLDB's test is not added into ctest, so coverage task will not run it from this pr.  
Just checked the it is the same number of tests runs by coverage compared to previous coverage job: 65. And we will end up add more tests to coverage task
+ java sdk compile will fail if compile hybridse standalone, so the `JAVASDK_ENABLE` option should removed
+ ~~some java tests failed~~

### CODE REVIEW

**Comment by aceforeverd** [comment]
[github workflow reorganization](https://trello.com/c/upawl5mF/8-github-workflow-reorganization)

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -55,13 +55,15 @@ function(compile_test DIR)
             get_filename_component(TEST_TARGET_DIR ${RELATIVE_TEST_PATH} DIRECTORY)
             get_filename_component(TEST_TARGET_NAME ${RELATIVE_TEST_PATH} NAME_WE)
             add_executable(${TEST_TARGET_NAME} ${SRC_FILE} $<TARGET_OBJECTS:openmldb_proto>)
+            list(APPEND test_list ${TEST_TARGET_NAME})
             if (${TEST_TARGET_NAME} STREQUAL "tablet_impl_mem_test")
```
Comment (aceforeverd) [inline src/CMakeLists.txt:59]: those test was not treat with `add_test()`, so coverage will not run

File: Makefile [hunk 2]
```diff
@@ -78,8 +80,15 @@ TEST_LEVEL ?=
 
 all: build
 
-# TODO(#677): add OpenMLDB coverage
-coverage: hybridse-coverage
+# TODO: better note about start zookeeper and onebox
+# some of the tests require zookeeper and openmldb server started before: checkout .github/workflows/coverage.yml
+coverage: coverage-configure
+	$(CMAKE_PRG) --build $(OPENMLDB_BUILD_DIR) -- -j$(NPROC)
```
Comment (aceforeverd) [inline Makefile]: rm this line

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Review by zhanghaohit — APPROVED** [review]
LGTM

**Comment by aceforeverd** [comment]
TODO: make coverage job runs faster

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**